sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -64,6 +66,22 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
 
   protected def sparkContext = sqlContext.sparkContext
 
+  /**
+   * ONE line description of this node.
+   */
+  override def simpleString(planLabelMap: mutable.LinkedHashMap[QueryPlan[_], Int]): String = {
+    s"$nodeName (${operatorIdStr(planLabelMap)})".trim
```

Comment:
It seems a little fragile to rely on QueryPlan equality to get the operator ID.

Reply:
@cloud-fan Using your idea to use tagging to store operator id will help here.